### PR TITLE
Add markCustomerOrder command and markSupplierOrder command

### PR DESCRIPTION
**Pull request by ryenl**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/MarkSupplierOrderCommand.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        // Retrieve the supplier order at the target index
+        SupplyOrder supplyOrder = supplierOrderList.getOrders().get(targetIndex - 1);
+
+        // Check if the order is already completed
+        if (supplyOrder.getStatus() == OrderStatus.COMPLETED) {
+            return new CommandResult(String.format(MESSAGE_ORDER_ALREADY_COMPLETED, targetIndex));
+        }
+
+        List<? extends Product> items = supplyOrder.getItems();  // List of items from the supplier order
+
```
Comment (calebchongsj) [inline src/main/java/seedu/address/logic/commands/MarkSupplierOrderCommand.java:50]: Nice use of generics for items in the supplier order.

**Review by calebchongsj — APPROVED** [review]
Looks good to me!

File: src/main/java/seedu/address/logic/commands/MarkCustomerOrderCommand.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import seedu.address.model.product.Ingredient;
+import seedu.address.model.product.Pastry;
+import seedu.address.model.product.Inventory;
+import seedu.address.model.product.Product;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MarkCustomerOrderCommand extends Command {
```
Comment (bbryant824) [inline src/main/java/seedu/address/logic/commands/MarkCustomerOrderCommand.java:17]: It seems abit messy here, may be you can try tidy the codes up for the next iteration.

**Review by bbryant824 — COMMENTED** [review]
Constructive feedback for Ryan.

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 3]
```diff
@@ -8,33 +8,7 @@
...
-import seedu.address.logic.commands.FilterCommand;
-import seedu.address.logic.commands.FindCommand;
-import seedu.address.logic.commands.HelpCommand;
-import seedu.address.logic.commands.ListCommand;
-import seedu.address.logic.commands.RemarkCommand;
-import seedu.address.logic.commands.ViewIngredientCatalogueCommand;
-import seedu.address.logic.commands.ViewOrderListCommand;
-import seedu.address.logic.commands.ViewPastryCatalogueCommand;
-import seedu.address.logic.commands.ViewInventoryCommand;
+import seedu.address.logic.commands.*;
```
Comment (samuelthen) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:11]: Is it better to list down all the commands for clarity purposes?

**Review by samuelthen — COMMENTED** [review]
Other parts look ok, but I think it is clearer to list all imports

File: src/main/java/seedu/address/logic/commands/MarkCustomerOrderCommand.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        if (customerOrder.getStatus() == OrderStatus.COMPLETED) {
+            return new CommandResult(String.format(MESSAGE_ORDER_ALREADY_COMPLETED, targetIndex));
+        }
+
+        List<? extends Product> items = customerOrder.getItems();
+
+        // Step 1: Accumulate required ingredient quantities
+        Map<Integer, Integer> requiredIngredients = new HashMap<>();
+
+        for (Product product : items) {
```
Comment (yikyak02) [inline src/main/java/seedu/address/logic/commands/MarkCustomerOrderCommand.java:56]: the nesting logic seems a bit deep here, is there any way to bypass the excessive use of nesting?

File: src/main/java/seedu/address/logic/commands/MarkSupplierOrderCommand.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import seedu.address.model.product.Ingredient;
+import seedu.address.model.product.Inventory;
+import seedu.address.model.product.Product;
+
+import java.util.List;
+
+public class MarkSupplierOrderCommand extends Command {
+    public static final String COMMAND_WORD = "markSupplierOrder";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Marks the supplier order at the given index as completed and updates the inventory.\n"
```
Comment (yikyak02) [inline src/main/java/seedu/address/logic/commands/MarkSupplierOrderCommand.java:17]: consider splitting this into more lines as it is rather long

**Pull request merged**